Orange/widgets/visualize/owmosaic.py

```diff
@@ -400,7 +400,7 @@ def set_data(self, data):
         self.closeContext()
         self.data = data
         self.init_combos(self.data)
-        if self.data is None or not len(self.data):
+        if self.data is None or not len(self.data) or not len(self.data[0]):
```

Comment:
I think it would be better to test this on domain, not data.

Besides, meta attributes are useful only if they are discrete or continuous, not strings. I think you should check this, too.